### PR TITLE
Create setting for UI URL and clarify Cloud API URL

### PR DESCRIPTION
**Pull request by madkinsz**
Changes `PREFECT_CLOUD_URL` to `PREFECT_CLOUD_API_URL` and adds `PREFECT_UI_URL` and `PREFECT_CLOUD_UI_URL`  to consolidate inference of UI URLs. 

`PREFECT_CLOUD_URL` is marked as deprecated and has backwards compatibility support.
`PREFECT_UI_URL` infers its value from `PREFECT_API_URL` by default, but can be set explicitly. For example, if you want to test against a UI preview on netlify.
`PREFECT_CLOUD_UI_URL` infers its value from `PREFECT_CLOUD_API_URL` by default, but can be set explicitly.

When using Cloud, the `PREFECT_UI_URL` is workspace scoped. The `PREFECT_CLOUD_UI_URL` is not.
When using the open source server, `PREFECT_CLOUD_UI_URL` will still point to Cloud. This is important for login e.g. #7334 

Required for #7334 
Helps with issues like https://github.com/PrefectHQ/prefect/pull/7173
### Example

```
❯ PREFECT_CLOUD_API_URL="bar" python -c "from prefect.settings import PREFECT_CLOUD_API_URL; print(PREFECT_CLOUD_API_URL.value())"
bar
```

```
❯ PREFECT_CLOUD_URL="foo" PREFECT_CLOUD_API_URL="bar" python -c "from prefect.settings import PREFECT_CLOUD_API_URL; print(PREFECT_CLOUD_API_URL.value())"
* Deprecation warning
foo
```

```
❯ PREFECT_CLOUD_API_URL="bar" python -c "from prefect.settings import PREFECT_CLOUD_URL; print(PREFECT_CLOUD_URL.value())" 
* Deprecation warning
bar
```
### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by madkinsz** [comment]
See also https://github.com/PrefectHQ/prefect/pull/7334/commits/1a2c92f7058a31ce9a6466389a3c3d0854766149 — which adds `PREFECT_CLOUD_UI_URL` for cases where the URL should not include the workspace

File: src/prefect/settings.py [hunk 1]
```diff
@@ -234,6 +234,43 @@ def warn_on_database_password_value_without_usage(values):
...
+    ui_url = api_url = PREFECT_API_URL.value_from(settings)
+
+    if not api_url:
+        return "http://ephemeral-orion"
+
+    if api_url.endswith("/api"):
+        ui_url = ui_url[:-4]
+
+    if api_url.startswith("https://api."):
+        ui_url = ui_url.replace("https://api.", "https://app.")
```
Comment (peytonrunyan) [inline src/prefect/settings.py:269]: Pure curiosity - is this a common issue?

Comment (madkinsz) [reply src/prefect/settings.py:269]: Hm idk about issue, this is just how we transform our API URl to a UI URL. I'm a little hesitant about having it be this broad honestly, maybe it should only replace `api.prefect.cloud` with `app.prefect.cloud`.

**Review by anticorrelator — COMMENTED** [review]
hey michael, will these changes require an update to the files [here](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/cli/orion_utils.py#L1)

**Comment by madkinsz** [comment]
Ah thanks for the note! Yeah we should probably look into changing those. Also looks like there are some additional transformations for Cloud that I'm missing :'( 

https://github.com/PrefectHQ/prefect/blob/main/src/prefect/cli/orion_utils.py#L76-L77

File: src/prefect/settings.py (hunk 1 above)
Comment (peytonrunyan) [reply src/prefect/settings.py:269]: Is there any downside to restricting it?

Comment (madkinsz) [reply src/prefect/settings.py:269]: Nope! I think we should. Otherwise we may incorrectly coerce someone's custom domain.

**Comment by madkinsz** [comment]
I've cherry-picked https://github.com/PrefectHQ/prefect/commit/1a2c92f7058a31ce9a6466389a3c3d0854766149 and updated handling to match the code linked by @anticorrelator

**Review by anticorrelator — APPROVED** [review]
lgtm!

**Pull request merged**